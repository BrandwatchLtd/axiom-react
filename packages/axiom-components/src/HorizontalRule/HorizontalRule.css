.ax-horizontal-rule {
  width: 100%;
  border-width: var(--component-border-width-small) 0 0 0;
  border-color: var(--color-theme-border);
}

.ax-horizontal-rule--solid {
  border-style: solid;
}

.ax-horizontal-rule--dotted {
  border-style: dotted;
}
