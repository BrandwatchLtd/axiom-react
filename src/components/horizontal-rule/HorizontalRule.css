.ax-horizontal-rule {
  width: 100%;
  margin: 0;
  border-width: var(--component-border-width-small) 0 0 0;
  border-color: var(--color-context-border);
}
