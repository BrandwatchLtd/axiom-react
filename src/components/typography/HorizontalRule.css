@import '../../materials/colors';
@import '../../materials/sizing';

.ax-horizontal-rule {
  border-top: var(--component-border-width) solid var(--color-context-border);
  border-right: 0;
  border-bottom: 0;
  border-left: 0;
  overflow: visible; /* Show the overflow in Edge. */
}
