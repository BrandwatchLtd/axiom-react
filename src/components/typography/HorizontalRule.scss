@import '../../design-patterns/colors/vars';
@import '../../design-patterns/spacing/vars';
@import '../../design-patterns/spacing/mixins';

.ax-horizontal-rule {
  @include space-block($component-margin-lg);

  border-top: $component-border-width solid $color-context-component-border;
  border-right: 0;
  border-bottom: 0;
  border-left: 0;
}
