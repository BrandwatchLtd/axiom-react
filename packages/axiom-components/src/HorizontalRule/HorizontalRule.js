import PropTypes from 'prop-types';
import classnames from 'classnames';
import React, { Component } from 'react';
import Base from '../Base/Base';
import './HorizontalRule.css';

export default class HorizontalRule extends Component {
  static propTypes = {
    borderStyle: PropTypes.oneOf(['solid', 'dotted']),
  };

  static defaultProps = {
    borderStyle: 'solid',
  };

  render() {
    const { borderStyle, ...rest } = this.props;
    const classes = classnames('ax-horizontal-rule', `ax-horizontal-rule--${borderStyle}`);

    return (
      <Base
          space="x0"
          { ...rest }
          Component="hr"
          className={ classes } />
    );
  }
}

