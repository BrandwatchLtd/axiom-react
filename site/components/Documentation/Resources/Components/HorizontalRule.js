import React, { Component } from 'react';
import { HorizontalRule } from '@brandwatch/axiom-components';
import {
  DocumentationApi,
  DocumentationContent,
  DocumentationShowCase,
} from '@brandwatch/axiom-documentation-viewer';

export default class Documentation extends Component {
  render() {
    return (
      <DocumentationContent>
        <DocumentationShowCase>
          <HorizontalRule />
        </DocumentationShowCase>

        <DocumentationApi components={ [
          require('!!axiom-documentation-loader!@brandwatch/axiom-components/src/HorizontalRule/HorizontalRule'),
        ] } />
      </DocumentationContent>
    );
  }
}
