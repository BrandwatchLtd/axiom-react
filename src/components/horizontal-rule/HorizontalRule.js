import React, { Component } from 'react';
import './HorizontalRule.css';
import Base from '../base/Base';

export default class HorizontalRule extends Component {
  render() {
    const { ...rest } = this.props;

    return (
      <Base space="x0" { ...rest } Component="hr" className="ax-horizontal-rule" />
    );
  }
}

