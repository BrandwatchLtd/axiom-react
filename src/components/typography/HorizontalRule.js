import React, { Component } from 'react';

if (__INCLUDE_CSS__) {
  require('./HorizontalRule.scss');
}

export default class HorizontalRule extends Component {
  render() {
    return (
      <hr className="ax-horizontal-rule" />
    );
  }
}
