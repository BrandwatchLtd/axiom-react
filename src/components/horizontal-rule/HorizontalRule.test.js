import React from 'react';
import renderer from 'react-test-renderer';
import HorizontalRule from './HorizontalRule';

const getComponent = (props = {}) =>
  renderer.create(
    <HorizontalRule { ...props } />
  );

describe('HorizontalRule', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with style', () => {
    ['solid', 'dotted'].forEach((borderStyle) => {
      it(borderStyle, () => {
        const component = getComponent({ borderStyle });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
