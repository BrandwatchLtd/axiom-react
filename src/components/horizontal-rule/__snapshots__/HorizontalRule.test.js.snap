// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalRule renders with defaultProps 1`] = `
<hr
  className="ax-horizontal-rule ax-horizontal-rule--solid ax-space--x0"
/>
`;

exports[`HorizontalRule renders with style dotted 1`] = `
<hr
  className="ax-horizontal-rule ax-horizontal-rule--dotted ax-space--x0"
/>
`;

exports[`HorizontalRule renders with style solid 1`] = `
<hr
  className="ax-horizontal-rule ax-horizontal-rule--solid ax-space--x0"
/>
`;
